pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -137,26 +147,50 @@ func (pfactory *PredicateMetadataFactory) GetMetadata(pod *v1.Pod, nodeNameToInf
 	if pod == nil {
 		return nil
 	}
-	// existingPodAntiAffinityMap will be used later for efficient check on existing pods' anti-affinity
-	existingPodAntiAffinityMap, err := getTPMapMatchingExistingAntiAffinity(pod, nodeNameToInfoMap)
+	// existingPodPredicateAntiAffinityMap will be used later for efficient check on existing pods' anti-affinity
+	// existingPodPriorityAffinityMap will be used later to calculate priority score on existing pods' anti-affinity and affinity
+	existingPodPredicateAntiAffinityMap, existingPodPriorityAffinityMap, err := getTPMapMatchingExistingAntiAffinity(pod, nodeNameToInfoMap, pfactory.hardPodAffinityWeight, pfactory.topologyInfo)
 	if err != nil {
 		return nil
 	}
 	// incomingPodAffinityMap will be used later for efficient check on incoming pod's affinity
 	// incomingPodAntiAffinityMap will be used later for efficient check on incoming pod's anti-affinity
-	incomingPodAffinityMap, incomingPodAntiAffinityMap, err := getTPMapMatchingIncomingAffinityAntiAffinity(pod, nodeNameToInfoMap)
+	// incomingPodPriorityAffinityMap will be used later to calculate priority score on incoming pods' anti-affinity and affinity
+	incomingPodPredicateAffinityMap, incomingPodPredicateAntiAffinityMap, incomingPodPriorityAffinityMap, err := getTPMapMatchingIncomingAffinityAntiAffinity(pod, nodeNameToInfoMap, pfactory.topologyInfo)
 	if err != nil {
 		klog.Errorf("[predicate meta data generation] error finding pods that match affinity terms: %v", err)
 		return nil
 	}
+
+	// use existingPodPriorityAffinityMap and incomingPodPriorityAffinityMap to calculate priority score for each node
+	nodeScore := make(map[string]int64, len(nodeNameToInfoMap))
+	if &pfactory.topologyInfo != nil && len(pfactory.topologyInfo) > 0 && existingPodPriorityAffinityMap != nil && incomingPodPriorityAffinityMap != nil {
```

Comment:
pfactory.topologyInfo check against nil is implied by len(pfactory.topologyInfo) > 0

Reply:
Done